dockstore-common/src/main/scala/io/dockstore/common/WdlBridge.scala

```diff
@@ -0,0 +1,412 @@
+import languages.wdl.draft3.WdlDraft3LanguageFactory
+import wdl.draft3.parser.WdlParser
+import wom.executable.WomBundle
+import cats.syntax.validated._
+import common.Checked
+import common.validation.Checked._
+import cromwell.languages.util.ImportResolver
+import languages.wdl.biscayne.WdlBiscayneLanguageFactory
+import languages.wdl.draft2.WdlDraft2LanguageFactory
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import wom.callable.{CallableTaskDefinition, WorkflowDefinition}
+import wom.expression.WomExpression
+import wom.graph.{ExternalGraphInputNode, OptionalGraphInputNode, OptionalGraphInputNodeWithDefault, RequiredGraphInputNode}
+import wom.types.{WomCompositeType, WomOptionalType, WomType}
+
+import scala.collection.JavaConverters
+
+
+/**
+  * A bridge class for interacting with the WDL documents
+  *
+  * Note: For simplicity, uses draft-3/v1.0 for throwing syntax errors from any language
+  */
+class WdlBridge {
+  var secondaryWdlFiles = new util.HashMap[String, String]()
+
+  def main(args: Array[String]): Unit = {
+    println("WdlBridge")
+  }
+
+  /**
+    * Set the secondary files (imports)
+    * @param secondaryFiles
+    */
+  def setSecondaryFiles(secondaryFiles: util.HashMap[String, String]): Unit = {
+    secondaryWdlFiles = secondaryFiles
+  }
+
+  /**
+    * Validates the workflow given by filePath
+    * @param filePath absolute path to file
+    */
+  @throws(classOf[WdlParser.SyntaxError])
+  def validateWorkflow(filePath: String) = {
+    val bundle = getBundle(filePath)
+
+    if (!bundle.primaryCallable.isDefined) {
+      throw new WdlParser.SyntaxError("Workflow is missing a workflow declaration.")
+    }
+  }
+
+  /**
+    * Validates the tool given by filePath
+    * @param filePath absolute path to file
+    */
+  @throws(classOf[WdlParser.SyntaxError])
+  def validateTool(filePath: String) = {
+    validateWorkflow(filePath)
+    val bundle = getBundle(filePath)
+    val executableCallable = bundle.toExecutableCallable.right.getOrElse(null)
+    if (executableCallable == null) {
+      throw new WdlParser.SyntaxError("Error parsing WDL file.")
+    }
+
+    if (executableCallable.taskCallNodes.seq.size > 1) {
+      throw new WdlParser.SyntaxError("A WDL tool can only have one task.")
+    }
+
+    executableCallable.taskCallNodes
+      .foreach(call => {
+        val dockerAttribute = call.callable.runtimeAttributes.attributes.get("docker")
+        if (!dockerAttribute.isDefined) {
+          throw new WdlParser.SyntaxError(call.identifier.localName + " requires an associated docker container to make this a valid Dockstore tool.")
+        }
+      })
+  }
+
+  /**
+    * Retrieves the metadata object for a given workflow
+    * @param filePath absolute path to file
+    * @throws wdl.draft3.parser.WdlParser.SyntaxError
```

Comment:
Is this true or will this sometimes throw wdl.draft3.parser.WdlParser.SyntaxError, wdl.v1(?).parser.WdlParser.SyntaxError, wdl.draft2.parser.WdlParser.SyntaxError, etc?

Reply:
I made it so that they all throw draft 3 error, since it doesn't really matter do to the abstraction of the version. I just chose draft 3 since it is the current version.